### ISSUE TITLE
Request body transform fails when template is loaded from file

### ISSUE THREAD

**Issue by NgariNdungu** [TykTechnologies/tyk#2518]
**Branch/Environment/Version**
- Branch/Version: 2.8.4
- Environment: On-prem

**Describe the bug**
Tyk fails to run the transform middleware for templates loaded from file.
```
error="failed to apply template to request: template: : \"\" is an incomplete or empty template"
```
The above is logged when a request is made to the endpoint, and the request body is passed to upstream unchanged.

**Reproduction steps**

1. Add api... 
2. Set up an endpoint with the body transform plugin:
- edit the raw api definition so that `extended_paths.transform` looks somewhat like;
```
"transform": [
              {
                "template_data": {
                  "input_type": "json",
                  "template_mode": "file",
                  "enable_session": false,
                  "template_source": "./templates/transform_test.tmpl",
                  "input": "",
                  "output": ""
                },
                "path": "/",
                "method": "POST"
              }
            ],
...
```
3. Place the template under the path defined in `template_source`
4. Restart tyk
5. Make request to defined endpoint as in [the docs](https://tyk.io/docs/transform-traffic/request-body/#json)

**Actual behavior**
The request body is passed on to the target url unchanged

**Expected behavior**
The template would be applied to transform the request body

**Comment by adelowo** [comment]
What is the content of `transform_test.tmpl`?

**Comment by NgariNdungu** [comment]
To add to this, my main suspect is [this line](https://github.com/TykTechnologies/tyk/blob/d80fd87800798c2746dc5f48621c5296cff14b51/gateway/api_definition.go#L572) in the `api_definition.go` source. The answer in this [thread](https://narkive.com/SMOmxKNv.6), suggests that calling `template.Execute()` on the returned template would raise the error referenced above.

**Comment by NgariNdungu** [comment]
> What is the content of `transform_test.tmpl`?

You can test with the tyk sample file [here](https://github.com/TykTechnologies/tyk/blob/master/templates/transform_test.tmpl)

**Comment by kgathi2** [comment]
Same issue here? Any solution for this bug?

**Comment by mkoppanen** [comment]
Seeing the same issue with templates loaded from files. Blob mode works just fine.

Big blocker at the moment. Is there a known working release version?

**Comment by mkoppanen** [comment]
I tested (docker) versions of tyk:

v2.8.5
v2.8.4
v2.8.3
v2.8.2
v2.8.1
v2.8.0

and ended up with same result. With a quick test 2.7.x and 2.5.x tags seem to have the same behaviour. This makes me think that my api definition must have an issue?

Let me know what information would help in debugging this issues.

**Comment by NgariNdungu** [comment]
This issue is still present in v2.9.3. @buger any timeline for when the change fixing this might be released?

**Comment by mkoppanen** [comment]
When I commented on the issue we were evaluating Tyk as a part of our platform. However, we ruled against choosing Tyk partially (large part) due to unresponsiveness to issues raised by the community.

**Comment by dunebuggie** [comment]
Hi @mkoppanen, I'm Shreyas, the community manager at Tyk. First off, thanks so much for the feedback. We know that we've not been as responsive as we can on this channel for sure. We're moving towards a new product and delivery approach, and community is a huge part of that. We're investing in our community to make sure we can support them, better. Apologies this ticket was missed. We'd love to speak to you about great communities you've been a part of to get some insight and your take on what makes a great OS community, would you be open to that?

**Comment by mkoppanen** [comment]
@dunebuggie,

Happy to assist you on that.

**Comment by anantasty** [comment]
@dunebuggie I am still experiencing the issue too. is there a timeline on a fix?

**Comment by dunebuggie** [comment]
> @dunebuggie,
> 
> Happy to assist you on that.

@mkoppanen, dropping you an email about this. Thanks so much! 


> @dunebuggie I am still experiencing the issue too. is there a timeline on a fix?

@anantasty We're reviewing it as part of the patch process and we're trying to get it included as quickly as possible, Anant. But in the meanwhile, if you're interested in contributing a bug fix, that's more than welcome. Will update here when this is fixed.

**Comment by kgathi2** [comment]
Any progress on this? The PR by @NgariNdungu , wasn't that merged?  #2661

**Comment by redstar01** [comment]
I'm faced with this bug too. Unfortunately, the this issue is still exist. (v2.9.4)

**Comment by marty-macfly** [comment]
Same with version 2.9.4.1

**Comment by moshemarciano** [comment]
same with v2.9.4.2, latest as of today.
The PR is marked as merged but isn't in the latest version, it is on master though.
@buger @dunebuggie @mkoppanen, reading templates from files is a basic functionality that is long since broken.. are community issues and PRs looked into? I'm evaluating Tyk to enhance our system but am now concerned

**Comment by furkansenharputlu** [comment]
This issue will be fixed in 3.0.1 release.
